plugins/player_auth/info.ts

```diff
@@ -1,24 +1,15 @@
 import * as lib from "@clusterio/lib";
 import * as messages from "./messages";
 
-class ControllerConfigGroup extends lib.PluginConfigGroup {}
-ControllerConfigGroup.defaultAccess = ["controller", "host", "control"];
-ControllerConfigGroup.groupName = "player_auth";
-ControllerConfigGroup.define({
-	name: "code_length",
-	title: "Code Length",
-	description: "Length in characters of the generated codes.",
-	type: "number",
-	initial_value: 6,
-});
-ControllerConfigGroup.define({
-	name: "code_timeout",
-	title: "Code Timeout",
-	description: "Time in seconds for the generated codes to stay valid.",
-	type: "number",
-	initial_value: 120,
-});
-ControllerConfigGroup.finalize();
+declare module "@clusterio/lib" {
+	export interface InstanceConfigFields {
+		"player_auth.load_plugin": boolean;
+	}
```

Comment:
It's automatically added to the field definitions, but it's not added to the type interface, so to read it you need to either use a cast or add it to the interface. Only player_auth actually reads this in its code so that is why this is the only place you see it added.

I don't think there's any sensible way to automatically add it to the type.